scipy/linalg/_matfuncs_sqrtm_triu.pyx

```diff
@@ -0,0 +1,33 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+import numpy as np
+from ._matfuncs_sqrtm import SqrtmError
+
+from numpy cimport complex128_t, float64_t, intp_t
+
+
+cdef fused floating:
```

Comment:
I haven't checked but are we expecting any esoteric numpy types e.g. int16 etc. here?

Reply:
The code in `_sqrtm_triu` casts the input to either of these two